sharedbuffers/mapped_struct.py

```diff
@@ -812,12 +819,27 @@ def unpack_from(cls, buf, offs, idmap = None):
         buf = _likerobuffer(buf)
         return cls(buf, offs, idmap)
 
-    @cython.locals(obj_offs = cython.ulonglong, dcode = cython.char, index = cython.ulonglong, pindex = "unsigned long *", dataoffs = cython.ulonglong)
-    def __getitem__(self, index):
+
+    @cython.locals(obj_offs = cython.ulonglong, dcode = cython.char, index = cython.longlong, pindex = "unsigned long *",
+        dataoffs = cython.ulonglong, objlen = cython.longlong, xlen = cython.longlong, step = cython.longlong)
+    def _getitem(self, index):
 
         dcode, objlen, itemsize, dataoffs, _struct = self._metadata()
+        xlen = objlen
+        step = 1
+
+        if self.elem_step != 0:
+            step = abs(self.elem_step)
+            if self.elem_step > 0:
+                xlen = (self.elem_end - self.elem_start - 1) / step + 1
+            else:
+                xlen = (self.elem_start - self.elem_end - 1) / step + 1
```

Comment:
This formula breaks if start==end

Reply:
Well, I disagree. If we went that route, we would have to check for buffer integrity everywhere for proxied objects, since they can be constructed that way too.

Regardless, I'll add the check since it's easy to do and not expensive at all.

Reply:
Well, yes. All classes do check for buffer integrity at risky places, AFAIK.